li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/LargeMessageHeaderValue.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * | Type   | UUID     | segmentNumber | numberOfSegments |
+ * | 1 byte | 16 bytes | 4 bytes       | 4 bytes          |
+ *
+ * The Type and UUID will be used to support large messages eventually.
+ * (as opposed to avro based large message serialization/deserialization)
+ */
+public class LargeMessageHeaderValue {
+  private final Type _type;
+  private final UUID _uuid;
+  private final int _segmentNumber;
+  private final int _numberOfSegments;
+
+  public LargeMessageHeaderValue(Type type, UUID uuid, int segmentNumber, int numberOfSegments) {
+    _type = type;
+    _uuid = uuid;
+    _segmentNumber = segmentNumber;
+    _numberOfSegments = numberOfSegments;
+  }
+
+  public int getSegmentNumber() {
+    return _segmentNumber;
+  }
+
+  public int getNumberOfSegments() {
+    return _numberOfSegments;
+  }
+
+  public UUID getUuid() {
+    return _uuid;
+  }
+
+  public Type getType() {
+    return _type;
+  }
+
+  public static byte[] toBytes(LargeMessageHeaderValue largeMessageHeaderValue) {
+    // 1 byte for type
+    // 16 bytes for UUID
+    // 4 bytes for segment number
+    // 4 bytes for number of segments
+    byte[] serialized = new byte[25];
+    serialized[0] = largeMessageHeaderValue.getType().getValue();
+    PrimitiveEncoderDecoder.encodeLong(largeMessageHeaderValue.getUuid().getLeastSignificantBits(), serialized, 1);
+    PrimitiveEncoderDecoder.encodeLong(largeMessageHeaderValue.getUuid().getMostSignificantBits(), serialized, 1 + PrimitiveEncoderDecoder.LONG_SIZE);
+    PrimitiveEncoderDecoder.encodeInt(
+        largeMessageHeaderValue.getSegmentNumber(),
+        serialized,
+        1 + 2 * PrimitiveEncoderDecoder.LONG_SIZE
+    );
+    PrimitiveEncoderDecoder.encodeInt(
+        largeMessageHeaderValue.getNumberOfSegments(),
+        serialized,
+        1 + 2 * PrimitiveEncoderDecoder.LONG_SIZE + PrimitiveEncoderDecoder.INT_SIZE
+    );
+    return serialized;
+  }
+
+  public static LargeMessageHeaderValue fromBytes(byte[] bytes) {
+    Type type = typeOf(bytes[0]);
+    long leastSignificantBits = PrimitiveEncoderDecoder.decodeLong(bytes, 1);
+    long mostSignificantBits = PrimitiveEncoderDecoder.decodeLong(bytes, 1 + PrimitiveEncoderDecoder.LONG_SIZE);
+    int segmentNumber = PrimitiveEncoderDecoder.decodeInt(bytes, 1 + 2 * PrimitiveEncoderDecoder.LONG_SIZE);
+    int numberOfSegments = PrimitiveEncoderDecoder.decodeInt(bytes, 1 + 2 * PrimitiveEncoderDecoder.LONG_SIZE + PrimitiveEncoderDecoder.INT_SIZE);
+    return new LargeMessageHeaderValue(type, new UUID(mostSignificantBits, leastSignificantBits), segmentNumber, numberOfSegments);
+  }
+
+  private static Type typeOf(byte value) {
+    switch (value) {
+      case (byte) 0: return Type.LEGACY;
+      default: throw new IllegalArgumentException("Unknown type found! = " + value);
+    }
+  }
+
+  public enum Type {
+    // This represents that both large message segments and encryption use avro serialization/deserialization
```

Comment:
the encryption/decryption interface is purely exposed to linkedin specific client. So, the comment and the implied meaning of this byte seems out of place. 

What is the "type" field in the header trying to represent? Is it the serialization format being used for large message ? Or is it used to represent if the payload is encrypted or not?  It can't imply both. 
I would suggest using bit-masks, if necessary to indicate different flags/indicators.

Reply:
I have removed the Type and made it a variable. We can add more variables with specific values as and when we implement the features.